pkg/orchestrator/etcd_worker.go

```diff
@@ -151,6 +156,9 @@ func (worker *EtcdWorker) Run(ctx context.Context, session *concurrency.Session,
 			return ctx.Err()
 		case <-sessionDone:
 			return cerrors.ErrEtcdSessionDone.GenWithStackByArgs()
+		case <-watchTicker.C:
+			log.Warn("watchCh blocking too long, reset watchCh")
+			watchCh = worker.client.Watch(ctx1, worker.prefix.String(), clientv3.WithPrefix(), clientv3.WithRev(worker.revision+1))
```

Comment:
Please also reset `ctx1` and call `cancel()` to release resources.

Reply:
Should we increase `worker.revision`?

Reply:
> 

I think the way now is fine.